lib/index.js

```diff
@@ -213,7 +213,9 @@ Segment.prototype.normalize = function(msg) {
       ctx.traits.crossDomainId = crossDomainId;
     }
   }
-  if (query) ctx.campaign = utm(query);
+  if (query && !ctx.campaign) {
```

Comment:
Thanks —  adding a comment certainly doesn't hurt, just wasn't sure if it was required! No need to remove it now :)